pkg/replicate/scheme.go

```diff
@@ -181,22 +180,74 @@ func (rs *replicationScheme) execute(ctx context.Context) error {
 		level.Info(rs.logger).Log("msg", "block meta not uploaded yet. Skipping.", "block_uuid", id.String())
 	}
 
+	filteredMetas := map[ulid.ULID]*metadata.Meta{}
 	for id, meta := range metas {
 		if rs.blockFilter(meta) {
 			level.Info(rs.logger).Log("msg", "adding block to be replicated", "block_uuid", id.String())
-			availableBlocks = append(availableBlocks, meta)
+			filteredMetas[id] = meta
 		}
 	}
 
-	// In order to prevent races in compactions by the target environment, we
-	// need to replicate oldest start timestamp first.
-	sort.Slice(availableBlocks, func(i, j int) bool {
-		return availableBlocks[i].BlockMeta.MinTime < availableBlocks[j].BlockMeta.MinTime
-	})
+	const (
+		deleteDelay = time.Duration(48 * time.Hour)
+	)
+	var (
+		compactMetrics                = newCompactMetrics(reg, deleteDelay)
+		acceptMalformedIndex          = false
+		enableVerticalCompaction      = false
+		blockFilesConcurrency         = 1
+		compactBlocksFetchConcurrency = 1
+	)
+
+	grouper := compact.NewDefaultGrouper(
```

Comment:
Compaction needs to group blocks together but replication doesn't have to. I think this is unnecessary

Reply:
As mentioned in the comments of the PRs raised earlier [1](https://github.com/thanos-io/thanos/pull/4350#discussion_r666064733) and [2](https://github.com/thanos-io/thanos/pull/5280#pullrequestreview-993151893) for the same issue, it was mentioned to create groups first. So, I have followed the same approach.